git-extra/git-update-git-for-windows

```diff
@@ -4,51 +4,131 @@
 # release.  If versions differ, the bit matched installer is downloaded and run
 # when confirmation to do so is given.
 
+#
+# Helpers for version_compare
+#
+numerical_compare () {
+	a1="$1"
+	b1="$2"
+
+	if test -z "$b1"
+	then
+		if test -z "$a1"
+		then
+			echo 0
+		else
+			echo 1
+		fi
+		return 1
+	fi
+	test -n "$a1" || { echo -1; return 1; }
+	test $a1 -le $b1 || { echo 1; return 1; }
+	test $b1 -le $a1 || { echo -1; return 1; }
+
+	return 0
+}
+
+lex_compare () {
+	a1="$1"
+	b1="$2"
+
+	if test -z "$b1"
+	then
+		if test -z "$a1"
+		then
+			return 0
+		else
+			echo 1
+		fi
+		return 1
+	fi
+	if [[ "x$a1" < "x$b1" ]]
+	then
+		echo -1
+		return 1
+	fi
+	if [[ "x$b1" < "x$a1" ]]
+	then
+		echo 1
+		return 1
+	fi
+
+	return 0
+}
```

Comment:
Hmm. I dislike the complexity of having to do lexicographic comparisons... Let me try whether I can come up with an alternative solution real quick.

Reply:
Sorry, I meant to comment immediately but forgot to "finalize my review".